### PR TITLE
CMake: suppress alignment warning for ARM64-MSVC

### PR DESCRIPTION
**Pull request by zheltovs**
Signed-off-by: Zheltov, Sergey1 <sergey1.zheltov@intel.com>

### CODE REVIEW

**Review by pavelkumbrasev — COMMENTED** [review]
There is no possibility that someone will get an error because of the substitution of this flag in the compilation?

File: cmake/compilers/MSVC.cmake [hunk 1]
```diff
@@ -21,6 +21,9 @@ string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 # Warning suppression C4324: structure was padded due to alignment specifier
 set(TBB_WARNING_LEVEL $<$<NOT:$<CXX_COMPILER_ID:Intel>>:/W4> $<$<BOOL:${TBB_STRICT}>:/WX>)
+if ("${MSVC_CXX_ARCHITECTURE_ID}" STREQUAL "ARM64")
+    set(TBB_ALIGNMENT_WARNING_SUPPRESS /wd4324)
+endif()
 set(TBB_WARNING_SUPPRESS /wd4324)
```
Comment (alexey-katranov) [inline cmake/compilers/MSVC.cmake:27]: Consider the following approach:
```suggestion
set(TBB_ALIGNMENT_WARNING_SUPPRESS /wd4324)
if ("${MSVC_CXX_ARCHITECTURE_ID}" STREQUAL "ARM64")
    set(TBB_PUBLIC_WARNING_SUPPRESS ${TBB_ALIGNMENT_WARNING_SUPPRESS})
else()
    set(TBB_WARNING_SUPPRESS ${TBB_ALIGNMENT_WARNING_SUPPRESS})
endif()
```

File: cmake/compilers/MSVC.cmake [hunk 2]
```diff
@@ -21,6 +21,9 @@ string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 # Warning suppression C4324: structure was padded due to alignment specifier
```
Comment (alexey-katranov) [inline cmake/compilers/MSVC.cmake:22]: Move this line to be just before `set(TBB_WARNING_SUPPRESS /wd4324)`

**Review by alexey-katranov — APPROVED** [review]
Consider updating release notes.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -33,6 +33,7 @@ function(tbb_add_test)
         PRIVATE
         ${TBB_CXX_STD_FLAG}
         ${TBB_WARNING_LEVEL}
+        ${TBB_ALIGNMENT_WARNING_SUPPRESS}
```
Comment (AlexVeprev) [inline test/CMakeLists.txt:36]: Just as idea how to get rid of this variable:
```suggestion
        # Warning suppression C4324 for ARM64: structure was padded due to alignment specifier
        $<$<STREQUAL:${MSVC_CXX_ARCHITECTURE_ID},ARM64>:/wd4325>
```

Is not it too complicated? :)

Comment (zheltovs) [reply test/CMakeLists.txt]: It really looks better, thanks for the suggestion!

**Pull request merged**